### DISCUSSION TITLE
Encrypted telnet connection with authentication - SSH support?

### DISCUSSION THREAD

**Discussion by SeimSoft** [daniele77/cli#109] [Ideas / Feature requests]
Hello,
the telnet server is pretty much insecure and I would need a way of a secure communication with the cli shell over the network.
Can you think of a way of doing this?
BR

**Comment by daniele77** [comment]
Hi, you can see the first part of issue #57 where a user asked about the authentication feature.

Basically, I answered:

> Currently, we use the library as a debug mechanism in embedded software: we faced the security problem in this way:
> 
> - the cli interface is bound to localhost
> - we ssh into the board
> - we telnet into the cli from localhost

However, you can consider becoming a sponsor of this project (click the heart button on top of the page!) to support the development of SSH feature in the library :-)
